upf/upf_api.c

```diff
@@ -1061,6 +1061,68 @@ vl_api_upf_tdf_ul_table_add_t_handler (vl_api_upf_tdf_ul_table_add_t * mp)
   REPLY_MACRO (VL_API_UPF_TDF_UL_TABLE_ADD_REPLY);
 }
 
+/* API message handler */
+static void
+vl_api_upf_ueip_pool_nwi_add_t_handler (vl_api_upf_ueip_pool_nwi_add_t * mp)
+{
+  int rv = 0;
+  upf_main_t *sm = &upf_main;
+  vl_api_upf_ueip_pool_nwi_add_reply_t *rmp = NULL;
+
+  u8 *nwi_name = mp->nwi_name;
+  u8 *nwi_name_vec = 0;
+  u8 *identity = mp->identity;
+  u8 *identity_vec = 0;
+  u8 identity_len = strlen (identity);
```

Comment:
Using `strlen` on a field that is potentially non-null-terminated is not correct